src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1396,6 +1404,42 @@ class DocBaseViewer extends BaseViewer {
             default:
         }
     }
+
+    /**
+     * Gets the cached thumbnails toggled state based on file id. Will retrieve from
+     * localStorage if not cached. Defaults to true if not found.
+     * @return {boolean} Whether thumbnails is toggled open or not
+     */
+    getCachedThumbnailsToggledState() {
+        let toggledOpen = true;
+
+        if (this.cache.has(THUMBNAILS_SIDEBAR_TOGGLED_MAP_KEY)) {
+            const thumbnailsToggledMap = this.cache.get(THUMBNAILS_SIDEBAR_TOGGLED_MAP_KEY);
+            toggledOpen = thumbnailsToggledMap[this.options.file.id];
+
+            // If cached toggled state is anything other than false, set it to true
+            if (toggledOpen !== false) {
+                toggledOpen = true;
+            }
+        }
+
+        return toggledOpen;
+    }
+
+    /**
+     * Caches the toggled state of the thumbnails sidebar, also saving to localStorage
+     * @param {boolean} isOpen Toggled state of the sidebar
+     * @return {void}
+     */
+    cacheThumbnailsToggledState(isOpen) {
+        let thumbnailsToggledMap = {};
+        if (this.cache.has(THUMBNAILS_SIDEBAR_TOGGLED_MAP_KEY)) {
```

Comment:
Do we even need to check this before? We're wiping it whether it exists or not.

Reply:
The cache entry is an object where each key is a file id, just like the current page map cache entry. So we want to get the existing cache entry and then extend it for this fileId

Reply:
It might read a bit better as below, since it looks like `Cache` checks for key presence already:

```
const thumbnailsCacheMap = this.cache.get(THUMBNAILS_SIDEBAR_TOGGLED_MAP_KEY) || {};
```